### PR TITLE
engineering: singleton design pattern in JS

### PR DESCRIPTION
**Pull request by chinhld12**
#### What's this PR do?

- [x] Singleton design pattern for using in global context

### CODE REVIEW

**Review by janrvu — CHANGES_REQUESTED** [review]
Please update the date (YYYY - MM - DD)

File: Engineering/Frontend/Singleton Design Pattern in Javascript.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+### Pros
+
+- You can be sure that a class has only a single instance.
+- A global access point to that instance.
+- The singleton only initialized once.
+
+### Cons
+
+- The Singleton is violate to the Single Responsibility Principle, via the SRP definition that a class only should only have one responsibility. But with the singleton pattern, it can carry too many responsibilities at the same time, this can make a deadlock when used in concurrency / multi-threads.
```
Comment (tienan92it) [inline Engineering/Frontend/Singleton Design Pattern in Javascript.md:97]: Another cons of singleton design pattern is difficult to write unit test cases.

**Review by tienan92it — APPROVED** [review]
LGTM

**Review by monotykamary — APPROVED** [review]
LGTM

**Pull request merged**